### ISSUE TITLE
Left-click position on map view not updated whith horizontal scroll

### ISSUE THREAD

**Issue by jakmeier** [jakmeier/paddlers-browser-game#8]
### Bug Description
After scrolling (x-axis) on the map view, the left-click positions are wrong. It is therefore impossible to select villages that are not visible without scrolling and to select 
### Possible Solution
Make sure the scrolled x offset is taken into account when looking up the village below the mouse.
Left clicks in the map view are dispatched here:
[paddlers-frontend/src/game/map/map_frame.rs](../tree/master/paddlers-frontend/src/game/map/map_frame.rs)
### Screenshots
![Screenshot_2020-05-14 Paddlers](https://user-images.githubusercontent.com/6342444/81954207-dd9a4080-9608-11ea-9a9e-88ff8ce5f3ff.jpg)

**Comment by GentileFulvio** [comment]
I'd like to have a look at this. I'm a rust noob though so this could be a challenge

**Comment by jakmeier** [comment]
Sounds awesome! I can give you some directions and you can ask me when you get stuck, that's always a good way of learning a new language.

In terms of coding, I think all we need is that the `x_offset` stored in the `struct GlobalMapSharedState` is applied when the click is checked, which happens in the function `map_position_lookup` in the file paddlers-frontend/src/game/map/map_position.rs

But first, we will have to see if the build system needs some updates, since you probably want to test it locally. It has been a while since my last clean checkout and so far I was the only coder in the project, so be warned that it probalby doesn't work out of the box. 

In theory, you should be able to run everything in docker without much trouble. If you have docker, docker-compose, and make installed, you can try to just run `make` in the root of the project and wait for a while until all projects have been built, on my PC that's all I have to do and it works. (send all errors to me if it doesn't work). If all goes well, you can then open a browser and you should see a login screen at localhost:80.

About the build times, on my laptop a clean build probably takes about an hour. Rebuilding and deploying the client takes only about s minute when you use the `make recreate-frontend` command, which will only shutdown one docker container and leave the others.
But a minute is still long to wait, therefore I always use [cargo-web](https://github.com/koute/cargo-web) outside of docker to do a `cargo web check` (from the frontend directory, not the project root!) before redeploying.

Sorry, I should really add all of this to the readme  but you could help me a lot by testing this first. 

If you have questions, let me know. We can discuss here on github or if you want a private channel, email should be okay to get us started. (jakob@paddlers.ch)

**Comment by GentileFulvio** [comment]
@jakmeier Thanks for the info! I'll get into it once i've got some time on my hands. 

I do have another question, i wanted to try out the demo but it stopped at `downloading buildings` and it didn't do anything after that.

I did try it on my phone so not sure if that is what caused the issue

**Comment by jakmeier** [comment]
Yes sure, try it when you got some spare time :)

About the demo, it turns out the game servers were overloaded this afternoon (I didn't expect so many people ti read my post) and now at least one of the databases seems to be down. I'll let you know when it's online again.

**Comment by jakmeier** [comment]
It's working again, you should now be able to log in. But I will need to look into stabilizing this...

**Comment by jakmeier** [comment]
Hey @GentileFulvio , short question, are you still interested in this issue? If yes, do you have a rough idea when you want to work on it? I'm planning to do a 0.1.3 release and update the live-demo in the not so far future, so I was wondering if I should wait for this.

**Comment by jakmeier** [comment]
@GentileFulvio 
I looked at this bug today because it really is an annoying bug and I wanted it to be fixed in the next update. Once I gave it a closer look and I found which numbers exactly needed to be changed, the fix was quite simple, so I implemented it, tested it and pushed it with 37b427c.

If you have some spare time at some point in the future and you are still interested in learning rust through contributing to this repo, let me know and I will find something a bit more interesting and with less code reading required. This issue probably wasn't the best for a rust beginner anyway, as the project code base is getting quite large and it can be hard to track down small bugs like these.
